### PR TITLE
Add linting to first-party polyfills

### PR DESCRIPTION
**Pull request by JakeChampion**
This pull-request adds linting to the polyfills included in the repository and fixes any warnings we had within the codebase.

Any polyfills which get turned into third-party polyfills will automatically be ignored by the linter.

### CODE REVIEW

**Comment by triblondon** [comment]
How about third party polyfills whose sources we are currently hosting?  I wouldn't want to cause test failures just by upgrading them from the original source.

File: polyfills/requestAnimationFrame/polyfill.js [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
+	if (rafPrefix) {
 		global.requestAnimationFrame = function (callback) {
-		    return webkitRequestAnimationFrame(function () {
+		    return global[rafPrefix + 'RequestAnimationFrame'](function () {
 		        callback(performance.now());
 		    });
 		};
-		global.cancelAnimationFrame = webkitCancelAnimationFrame;
-
+		global.cancelAnimationFrame = global[rafPrefix + 'mozCancelAnimationFrame'];
```
Comment (rowanmanning) [inline polyfills/requestAnimationFrame/polyfill.js]: Won't this sometimes end up being `webkitmozCancelAnimationFrame`?

**Comment by rowanmanning** [comment]
One comment, otherwise this looks great to me 👍

**Comment by JakeChampion** [comment]
@triblondon Do you happen to know which polyfills those would be?

**Comment by triblondon** [comment]
All the ones with `repo` properties in their config.json files.

**Comment by JakeChampion** [comment]
@rowanmanning Happy for this to be merged in?

**Comment by rowanmanning** [comment]
Yep, looks good 👍

**Pull request merged**